php/Blockonomics.php

```diff
@@ -917,6 +917,10 @@ public function process_callback($secret, $address, $status, $value, $txid, $rbf
         }
 
         $this->update_order($order);
+
+        $blockonomics_currencies = $this->getSupportedCurrencies();
+        $selected_currency = $blockonomics_currencies[$order['crypto']];
+        $wc_order->set_payment_method_title($selected_currency['name']);
```

Comment:
Here we've found an edge case. Where if a customer does a partial payment with 2 different transactions of both BTC & BCH, then with the above logic, we'll be showing the payment method used in the last transaction.

To fix it, we'll need to make another query to fetch all the past transactions of the order and then compute the final title based on the different currency used.

Question is if it's worth doing an extra query for this edge case.

Reply:
```php
global $wpdb;

$query = $wpdb->prepare(
    "SELECT crypto FROM ".$wpdb->prefix."blockonomics_payments WHERE order_id = %d AND txid != ''",
    $order['order_id']
);

$results = $wpdb->get_results($query, ARRAY_A);

$blockonomics_currencies = $this->getSupportedCurrencies();
$selected_currency = $blockonomics_currencies[$order['crypto']];
$payment_method_title = $selected_currency['name'];

if ($results) {
    $cryptos = array_column($results, 'crypto');
    $cryptos[] = $order['crypto'];
    
    $cryptos_unique = array_unique($cryptos);

    if (in_array('btc', $cryptos_unique) && !in_array('bch', $cryptos_unique)) {
        $payment_method_title = $blockonomics_currencies['btc']['name'];
    } elseif (!in_array('btc', $cryptos_unique) && in_array('bch', $cryptos_unique)) {
        $payment_method_title = $blockonomics_currencies['bch']['name'];
    } elseif (in_array('btc', $cryptos_unique) && in_array('bch', $cryptos_unique)) {
        $payment_method_title = $blockonomics_currencies['btc']['name'] . ", " . $blockonomics_currencies['bch']['name'];
    }
}

$wc_order->set_payment_method_title($payment_method_title);
```